tests/test_modeling_layoutlmv2.py

```diff
@@ -0,0 +1,566 @@
+    image = ImageList(torch.zeros(13, 3, 256, 256), 256)
+    position_ids = token_type_ids
+    # fmt: on
+
+    return input_ids, attention_mask, bbox, image, token_type_ids, position_ids, labels
+
+
+@require_torch
+class LayoutLMv2ModelIntegrationTest(unittest.TestCase):
+    @slow
+    def test_forward_pass_no_head(self):
+        model = LayoutLMv2Model.from_pretrained("microsoft/layoutlmv2-base-uncased").to(torch_device)
+
+        (
+            input_ids,
+            attention_mask,
+            bbox,
+            image,
+            token_type_ids,
+            position_ids,
+            labels,
+        ) = prepare_layoutlmv2_batch_inputs()
+
+        # forward pass
+        outputs = model(
+            input_ids=input_ids,
+            bbox=bbox,
+            image=image,
+            attention_mask=attention_mask,
+            token_type_ids=token_type_ids,
+            position_ids=position_ids,
+        )
+
+        # test the sequence output on [0, :3, :3]
+        expected_slice = torch.tensor(
+            [[0.1785, -0.1947, -0.0425], [-0.3254, -0.2807, 0.2553], [-0.5391, -0.3322, 0.3364]],
+            device=torch_device,
+        )
+
+        self.assertTrue(torch.allclose(outputs.last_hidden_state[0, :3, :3], expected_slice, atol=1e-3))
+
+        # test the pooled output on [1, :3]
+        expected_slice = torch.tensor([-0.6580, -0.0214, 0.8552], device=torch_device)
+
+        self.assertTrue(torch.allclose(outputs.pooler_output[1, :3], expected_slice, atol=1e-3))
+
+    @slow
+    def test_forward_pass_token_classification(self):
+        # initialize model with randomly initialized token classification head
+        model = LayoutLMv2ForTokenClassification.from_pretrained("microsoft/layoutlmv2-base-uncased", num_labels=3).to(
+            torch_device
+        )
+
+        (
+            input_ids,
+            attention_mask,
+            bbox,
+            image,
+            token_type_ids,
+            position_ids,
+            labels,
+        ) = prepare_layoutlmv2_batch_inputs()
+
+        # forward pass
+        outputs = model(
+            input_ids=input_ids,
+            bbox=bbox,
+            image=image,
+            attention_mask=attention_mask,
+            token_type_ids=token_type_ids,
+            position_ids=position_ids,
+        )
+
+        # test the loss calculation to be around 2.65
+        # expected_loss = torch.tensor(2.65, device=torch_device)
+
+        # The loss is currently somewhat random and can vary between 0.1-0.3 atol.
+        # self.assertTrue(torch.allclose(outputs.loss, expected_loss, atol=0.1))
+
+        # test the shape of the logits
+        logits = outputs.logits
+        expected_shape = torch.Size((2, 25, 13))
+        self.assertEqual(logits.shape, expected_shape)
```

Comment:
Can you also add a test that checks a slice of the logits (using `torch.allclose`)?